File: java/lasagna/src/main/java/Lasagna.java
class Main {
    public static void main(String[] args) {
        // Your code goes here
        Lasagna lasagna = new Lasagna();
        lasagna.expectedMinutesInOven();
        lasagna.remainingMinutesInOven(25);
        lasagna.preparationTimeInMinutes(1);
        lasagna.preparationTimeInMinutes(4);
        lasagna.totalTimeInMinutes(1, 30);
        lasagna.totalTimeInMinutes(4, 8);
    }
}




public class Lasagna {
    // TODO: define the 'expectedMinutesInOven()' method
    private int expectedTimeLasagnaInOven = 40;
    public int expectedMinutesInOven(){
        /*
         does not take any parameters and returns how many minutes the lasagna
         should be in the oven.
         According to the cooking book, the expected oven time in minutes is 40
        * */
        return expectedTimeLasagnaInOven;
    }

    // TODO: define the 'remainingMinutesInOven()' method
    /*
    *takes the actual minutes the lasagna has been in the oven as a parameter and returns how many
    * minutes the lasagna still has to remain in the ove
    * */
    public int remainingMinutesInOven(int MinutesLasagnaHasBeenInOven){

        return expectedTimeLasagnaInOven - MinutesLasagnaHasBeenInOven;
    }

    // TODO: define the 'preparationTimeInMinutes()' method
    /*
    takes the number of layers you added to the lasagna as a parameter and returns how many minutes you spent preparing the lasagna,
    assuming each layer takes you 2 minutes to prepare
    * */
    public int preparationTimeInMinutes(int numerOfLayers){
        return numerOfLayers * 2;
    }

    // TODO: define the 'totalTimeInMinutes()' method
    /*
     that takes two parameters: the first parameter is the number of layers you added to the lasagna, and the second parameter is the
     number of minutes the lasagna has been in the oven. The function should return how many minutes in total you've worked on cooking
     the lasagna, which is the sum of the preparation time in minutes, and the time in minutes the lasagna has spent in the oven at the moment.
    * */
    public int totalTimeInMinutes(int numberOfLayers, int minutesLasagnaHasBeenOven){
        numberOfLayers *= 2;
        return numberOfLayers + minutesLasagnaHasBeenOven;
    }
}

File: python/guidos-gorgeous-lasagna/lasagna.py

#TODO (student): define your EXPECTED_BAKE_TIME (required) and PREPARATION_TIME (optional) constants below.
EXPECTED_BAKE_TIME = 40

#TODO (student): Remove 'pass' and complete the 'bake_time_remaining()' function below.
def bake_time_remaining(elapsed_bake_time):

    """
    bake_time_remaining
    Returns:
        int: The remaining bake time (in minutes) derived from 'EXPECTED_BAKE_TIME'.

    Function that takes the actual minutes the lasagna has been in the oven as
    an argument and returns how many minutes the lasagna still needs to bake
    based on the `EXPECTED_BAKE_TIME`.
    """
    return EXPECTED_BAKE_TIME - elapsed_bake_time
def preparation_time_in_minutes(number_of_layers):
    """preparation_time_in_minutes

        Parameters:
            number_of_layers (int): The number of layers in the lasagna.

        Returns:
            int: The total time elapsed (in minutes) preparing

        This function takes one integer representing the number of lasagna
        layers.
        It calculates the total elapsed minutes spent cooking.

        """
    return number_of_layers * 2

def elapsed_time_in_minutes(number_of_layers, elapsed_bake_time):
    """Calculate the elapsed cooking time.

        Parameters:
            number_of_layers (int): The number of layers in the lasagna.
            elapsed_bake_time (int): Time the lasagna has been baking in the oven.

        Returns:
            int: The total time elapsed (in minutes) preparing and baking.

        This function takes two integers representing the number of lasagna
        layers and the time already spent baking the lasagna. It calculates
        the total elapsed minutes spent cooking (preparing + baking).

        """
    return (number_of_layers * 2) + elapsed_bake_time

print(EXPECTED_BAKE_TIME)

bake_time_remaining(1)
bake_time_remaining(2)
bake_time_remaining(5)
bake_time_remaining(10)
bake_time_remaining(15)
bake_time_remaining(23)
bake_time_remaining(33)
bake_time_remaining(39)

preparation_time_in_minutes(1)
preparation_time_in_minutes(2)
preparation_time_in_minutes(5)
preparation_time_in_minutes(8)
preparation_time_in_minutes(11)
preparation_time_in_minutes(15)

elapsed_time_in_minutes(1, 3)
elapsed_time_in_minutes(2, 7)
elapsed_time_in_minutes(5, 8)
elapsed_time_in_minutes(8, 4)
elapsed_time_in_minutes(11, 15)
elapsed_time_in_minutes(15, 20)


